src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -3558,27 +3553,27 @@ void ride_construction_toolupdate_construct(ScreenCoordsXY screenCoords)
 
     gMapSelectArrowPosition.z = z;
     bx = 41;
-    _currentTrackBegin.x = x;
-    _currentTrackBegin.y = y;
+    _currentTrackBegin.x = mapCoords.x;
+    _currentTrackBegin.y = mapCoords.y;
     _currentTrackBegin.z = z;
-    if ((_currentTrackSelectionFlags & TRACK_SELECTION_FLAG_TRACK) && x == _previousTrackPiece.x && y == _previousTrackPiece.y
-        && z == _previousTrackPiece.z)
+    if ((_currentTrackSelectionFlags & TRACK_SELECTION_FLAG_TRACK) && mapCoords.x == _previousTrackPiece.x
```

Comment:
A comparison between currentTrackBegin and previousTrackPiece could simplify this instead of breaking it into the components.